Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/eventregistration/attendee/edit/EditCheckInFragment.kt

```diff
@@ -0,0 +1,171 @@
+import de.rki.coronawarnapp.util.ui.observe2
+import de.rki.coronawarnapp.util.ui.popBackStack
+import de.rki.coronawarnapp.util.ui.setGone
+import de.rki.coronawarnapp.util.ui.viewBindingLazy
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactoryProvider
+import de.rki.coronawarnapp.util.viewmodel.cwaViewModelsAssisted
+import org.joda.time.DateTimeZone
+import org.joda.time.LocalDate
+import org.joda.time.LocalTime
+import javax.inject.Inject
+
+class EditCheckInFragment : Fragment(R.layout.fragment_edit_check_in), AutoInject {
+    private val navArgs by navArgs<EditCheckInFragmentArgs>()
+
+    @Inject lateinit var viewModelFactory: CWAViewModelFactoryProvider.Factory
+    private val viewModel: EditCheckInViewModel by cwaViewModelsAssisted(
+        factoryProducer = { viewModelFactory },
+        constructorCall = { factory, _ ->
+            factory as EditCheckInViewModel.Factory
+            factory.create(if (navArgs.editCheckInId == 0L) null else navArgs.editCheckInId)
+        }
+    )
+
+    private val binding: FragmentEditCheckInBinding by viewBindingLazy()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        with(binding) {
+            toolbar.setNavigationOnClickListener { viewModel.onClose() }
+
+            editCheckinEditCardCheckinDate.setOnClickListener {
+                viewModel.onStartDateClicked()
+            }
+
+            editCheckinEditCardCheckinTime.setOnClickListener {
+                viewModel.onStartTimeClicked()
+            }
+
+            editCheckinEditCardCheckoutDate.setOnClickListener {
+                viewModel.onEndDateClicked()
+            }
+
+            editCheckinEditCardCheckoutTime.setOnClickListener {
+                viewModel.onEndTimeClicked()
+            }
+
+            editCheckinConfirmButton.setOnClickListener {
+                viewModel.onSaveClicked()
+            }
+        }
+
+        viewModel.events.observe2(this) { navEvent ->
+            when (navEvent) {
+                EditCheckInNavigation.BackNavigation -> popBackStack()
+                EditCheckInNavigation.ConfirmNavigation -> {
+                    popBackStack()
+                }
+            }
+        }
+
+        viewModel.uiState.observe2(this) { uiState ->
+            with(binding) {
+                editCheckinInfoCardHeader.text = getString(uiState.typeRes)
+                editCheckinInfoCardTitle.text = uiState.description
+                editCheckinInfoCardAddress.text = uiState.address
+
+                editCheckinEditCardCheckinDate.text = uiState.checkInStartDate
+                editCheckinEditCardCheckinTime.text = uiState.checkInStartTime
+
+                editCheckinEditCardCheckoutDate.text = uiState.checkInEndDate
+                editCheckinEditCardCheckoutTime.text = uiState.checkInEndTime
+
+                editCheckinDurationEditHintCard.setGone(!uiState.diaryWarningVisible)
```

Comment:
`isGone` from `ktx` ? hopefully we can get rid of DataBinding and its binding adapters